src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1320,54 +1387,67 @@ private void ImportThrow()
             EmitTrapCall();
         }
 
-        private void ImportLoadField(int token, bool isStatic)
+        private LLVMValueRef GetInstanceFieldAddress(StackEntry objectEntry, FieldDesc field)
         {
-            if (isStatic)
+            var objectType = objectEntry.Type ?? field.OwningType;
```

Comment:
Same question as before on whether we can just use field.OwningType

Reply:
I actually tried using just field.OwningType and it appears there are some scenarios where the field.OwningType != objectEntry.Type && objectEntry.Type != null. Do you want me to track down the specific scenarios where someone has pushed an expression with a different (but somehow right) type?

Reply:
You could see that with a case like:
```
class Base
{
    public int BaseField;
}
class Derived : Base
{
    public int DerivedField;
}
void SomeMethod()
{
    new Derived().BaseField = 3;
}
```

In that case, objectEntry.Type would be Derived and field.OwningType would be Base. They both are guaranteed to work -- derived types inherit the layout of their base class. Since we are missing some opcodes, we could get other funny-looking, but harmless results. For example:
```
void SomeOtherMethod(Base obj)
{
    ((Derived)obj).DerivedField = 4;
}
```
Would currently have objectEntry.Type set to Base because we haven't implemented casting. However, using the layout for Derived.DerivedField is still the correct choice.

Reply:
while the results at runtime may be harmless, LLVM pukes because the types don't exactly match, at least in the case I saw.

Reply:
Ah... I see now. I was expecting this to be used for the field offset, but we actually figure that out on line 1407. What you're doing here is right for cases like byrefs.